testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -5539,6 +5539,72 @@ private static void checkIf(SqlOperatorFixture f) {
     f.checkNull("atan2(1, cast(null as double))");
   }
 
+  @Test void testAcoshFunc() {
+    final SqlOperatorFixture f0 = fixture().setFor(SqlLibraryOperators.ACOSH);
+    f0.checkFails("^acosh(1)^",
+        "No match found for function signature ACOSH\\(<NUMERIC>\\)",
+        false);
+    final Consumer<SqlOperatorFixture> consumer = f -> {
+      f.checkType("acosh(1)", "DOUBLE NOT NULL");
+      f.checkType("acosh(cast(1 as Decimal))", "DOUBLE NOT NULL");
+      f.checkType("acosh(case when false then 1 else null end)", "DOUBLE");
+      f.checkType("acosh('abc')", "DOUBLE NOT NULL");
+      f.checkScalarApprox("acosh(1)", "DOUBLE NOT NULL",
+          isWithin(0d, 0.0001d));
+      f.checkScalarApprox("acosh(cast(10 as decimal))", "DOUBLE NOT NULL",
+          isWithin(2.9932d, 0.0001d));
+      f.checkNull("acosh(cast(null as integer))");
+      f.checkNull("acosh(cast(null as double))");
+      f.checkFails("acosh(0.1)",
+          "Input parameter of acosh cannot be small than 1!",
```

Comment:
How is this test passing? Shouldn't the expected error message be adapted?

Reply:
I found it is because we only check this expression will be successful during parse and validate. The error message has been ignored, you could see more details in https://github.com/apache/calcite/blob/f32289870be80e69dabc5b7e24bd6c777d4eb688/testkit/src/main/java/org/apache/calcite/test/SqlOperatorFixtureImpl.java#L161

Maybe we should improve this?

Reply:
Ah, ok. Thanks for the info. It is certainly counter-intuitive and probably there's room for improvement, but IMO this is out of the scope of the current PR.
I would say, please just align this "expected error message" with the actual error that is thrown, just for the sake of correctness (and in case the fixture mechanism gets improved in the future).